### ISSUE TITLE
PowerOn not working

### ISSUE THREAD

**Issue by rkokkelk** [arcreative/xbox-on#9]
Hi,

Great plugin, however recently it has stopped working on the Xbox-one, probably due to the recent firmware upgrade. I've verified it that a single UDP packet is not sufficient to start the booting procedure but multiple UDP packets will. For example, the Python version of this application [xbox-remote-power](https://github.com/Schamper/xbox-remote-power) does still work because it sends multiple packets instead of one. 

The reason that I prefer a fix for this is because your application is used by an add-on for the Athom Homey device which multiple people use including myself, [add-on page](https://apps.athom.com/app/net.weejewel.xboxone). So could you please look into this issue. 

Thank you!

**Comment by arcreative** [comment]
I had left this up to the user as an implementation detail... since the request is essentially synchronous and doesn't get a response.  My only concern in adding a repeat as default behavior is that implementations using this module are already doing the repeat on their end...

Can't you just do 
```
for (var i = 0; i < 5; i++) {
    setTimeout(xbox.powerOn, 100);
}
```
On the plugin's end?  Or is your concern that repeat packets can't be invoked from the CLI?

**Comment by rkokkelk** [comment]
Well I think that the assumption that people have by using your module is now incorrect. As for me for example, using the CLI or using the module `poweron` functionality did not result in the expected result, the Xbox turning on. 

Another example, another plugin is using your module and is issuing the`poweron` function only once, because it has the expectation that the Xbox will turned on by using the function only once. So in general I would expect that this functionality will be implemented within the module itself instead of it being relied on by the plugin.

I've modified the current changes because I thought that the source port of the UDP sockets had to remain the same for the poweron procedure to work. Because as you correctly mentioned it does not have to be. So if you agree with implementing the functionality within the module, I'll modify the PR request. 

If you don't agree, I will make a new PR so that the CLI will work and contact the developer of the add-on which is using your module that it needs to change its implementation.

Thank you!

**Comment by WeeJeWel** [comment]
I'm using the module, and I agree with @rkokkelk that the module should power on the Xbox. If the method was named 'sendPacket' or so, it would be ambiguous, but the naming of the current method implies it should simply work. 

Maybe you can add an object as parameter with a 'repetition' property, defaulting to 5 or so.

**Comment by arcreative** [comment]
Thanks all, I've published `0.0.9` / `376d778` per @rkokkelk's pull request.

I'm not sure I quite agree that Microsoft's shittifying of the Xbox's UDP sensitivity is really this module's fault, but I don't think there's any harm in sending extra dgrams either.  Xbox WOL has always been a poke-and-pray method at best (I still can't even get SmartGlass to do it), so all these implementations must be taken with a grain of salt :-)

In any case, thanks for the comments and let me know how the updates work out for you 🍻

**Comment by rkokkelk** [comment]
Thank you for updating the add-on. I've just tried the modifications and the default of 500ms is a little too short for my Xbox. But specifying the delay to 1s did the trick. So it works. 

Thank again!

**Comment by arcreative** [comment]
So weird... seems like even the weather affects these things.  FWIW it's always worked with 1 packet for me, so not sure what the deal is.  What model of Xbox One do you have and when did you get it?

**Comment by arcreative** [comment]
I pushed `0.0.10` to update default delay to 1000ms.  It seems that's what most are using anyway, just seems like a really long time to wait for the CLI to terminate.

**Comment by rkokkelk** [comment]
Great! I agree with you that it really depends on which xbox you are using. But for me 5 pkts with 1s delay resulted in the xbox powering up 100 percent of the time. 

Thank you again for the updates!
